From: Jeff Mahoney <jeffm@suse.com>
Subject: [PATCH] apparmor: work around new call of should_remove_suid()

 This patch works around a new call of should_remove_suid() in do_truncate()
 added after 2.6.21 was released.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>

---
 fs/open.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/fs/open.c	2007-05-25 17:56:01.000000000 -0400
+++ b/fs/open.c	2007-05-25 17:56:21.000000000 -0400
@@ -198,6 +198,7 @@ int do_truncate(struct dentry *dentry, s
 {
 	int err;
 	struct iattr newattrs;
+	struct path path = { .mnt = mnt, .dentry = dentry };
 
 	/* Not pretty: "inode->i_size" shouldn't really be signed. But it is. */
 	if (length < 0)
@@ -211,7 +212,7 @@ int do_truncate(struct dentry *dentry, s
 	}
 
 	/* Remove suid/sgid on truncate too */
-	newattrs.ia_valid |= should_remove_suid(dentry);
+	newattrs.ia_valid |= should_remove_suid(&path);
 
 	mutex_lock(&dentry->d_inode->i_mutex);
 	err = notify_change(dentry, mnt, &newattrs);
